.github/workflows/preview-indexer.yaml

```diff
@@ -0,0 +1,116 @@
+name: Generate/Updates Preview Index For DA Tenants
+
+on:
+  workflow_dispatch:
+    inputs:
+      lastRunISOTo:
+        description: "Last ISO To timestamp"
+        required: false
+        type: string
+      lastRunISOFrom:
+        description: "Last ISO From timestamp"
+        required: false
+        type: string
+      site:
+        description: "Site name e.g da-bacom"
+        required: false
+        type: string
+      siteRegions:
+        description: "Comma-separated regional paths e.g /au/,/de/"
+        required: false
+        type: string
+
+permissions:
+  actions: read # required to access workflow runs
+  contents: write # required to access the repository, for the dispatch event
+
+env:
+  ROLLING_IMPORT_IMS_URL: ${{ secrets.ROLLING_IMPORT_IMS_URL }}
+  ROLLING_IMPORT_CLIENT_ID: ${{ secrets.ROLLING_IMPORT_CLIENT_ID }}
+  ROLLING_IMPORT_CLIENT_SECRET: ${{ secrets.ROLLING_IMPORT_CLIENT_SECRET }}
+  ROLLING_IMPORT_CODE: ${{ secrets.ROLLING_IMPORT_CODE }}
+  ROLLING_IMPORT_GRANT_TYPE: ${{ secrets.ROLLING_IMPORT_GRANT_TYPE }}
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  PREVIEW_INDEXER_REPO: ${{ vars.PREVIEW_INDEXER_REPO }}
+  PREVIEW_INDEX_FILE: ${{ vars.PREVIEW_INDEX_FILE }}
+  AEM_ADMIN_TOKEN_ADOBECOM_DA_BACOM: ${{ secrets.AEM_ADMIN_TOKEN_ADOBECOM_DA_BACOM }}
+  EXCLUDE_PREVIEW_PATHS_ADOBECOM_DA_BACOM: ${{ vars.EXCLUDE_PREVIEW_PATHS_ADOBECOM_DA_BACOM }}
+  PREVIEW_ROOTS_ADOBECOM_DA_BACOM: ${{ vars.PREVIEW_ROOTS_ADOBECOM_DA_BACOM }}
+  PREVIEW_PATH_EXTN_ADOBECOM_DA_BACOM: ${{ vars.PREVIEW_PATH_EXTN_ADOBECOM_DA_BACOM }}
+
+jobs:
+  preview-indexer:
+    if: github.repository_owner == 'adobecom'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check if previous run is in progress
+        id: check-run
+        run: |
+          # If MOCK_GH_RUNNING is set, skip immediately
+          if [ "${MOCK_GH_RUNNING}" = "true" ]; then
+            echo "Mock mode: pretending another run is active. Skipping..."
+            exit 78
+          elif [ "${MOCK_GH_RUNNING}" = "false" ]; then
+            echo "Mock mode: pretending another run is not active. Skipping..."
+            exit 0
+          fi
+
+          # Otherwise check real GitHub runs (only works in actual Actions)
+          running=$(gh run list \
+            --workflow="${{ github.workflow }}" \
+            --branch="${{ github.ref_name }}" \
+            --status=in_progress \
+            --limit 1 \
+            --json status \
+            --jq 'length' || echo "0")
+
+          if [ "$running" -gt 0 ]; then
+            echo "Another run is in progress. Exiting..."
+            exit 78
+          fi
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
```

Comment:
could we use SHAs? They can't get changed thus you can pin a version, avoiding any supply chain attacks that could steal credentials https://docs.github.com/en/actions/reference/security/secure-use#using-third-party-actions